File: fsharp/isogram/IsogramTest.fs
module IsogramTest
    
open NUnit.Framework

open Isogram

[<TestCase("duplicates", ExpectedResult = true)>]
[<TestCase("eleven", ExpectedResult = false)>]
[<TestCase("subdermatoglyphic", ExpectedResult = true)>]
[<TestCase("Alphabet", ExpectedResult = false)>]
[<TestCase("thumbscrew-japingly", ExpectedResult = true)>]
[<TestCase("Hjelmqvist-Gryb-Zock-Pfund-Wax", ExpectedResult = true)>]
[<TestCase("Heizölrückstoßabdämpfung", ExpectedResult = true)>]
[<TestCase("the quick brown fox", ExpectedResult = false)>]
[<TestCase("Emily Jung Schwartzkopf", ExpectedResult = true)>]
[<TestCase("éléphant", ExpectedResult = false)>]
let ``Isogram correctly detects isograms`` (actual: string) =
    isogram actual

File: scala/isogram/src/test/scala/IsogramTest.scala
import org.scalatest.{FunSuite, Matchers}

class IsogramTest extends FunSuite with Matchers {

  test("empty string") {
    val result = Isogram.isIsogram("")
    result should be (true)
  }

  test("isogram with only lower case characters") {
    
    val result = Isogram.isIsogram("isogram")
    result should be (true)
  }

  test("word with one duplicated character") {
    
    val result = Isogram.isIsogram("eleven")
    result should be (false)
  }

  test("longest reported english isogram") {
    
    val result = Isogram.isIsogram("subdermatoglyphic")
    result should be (true)
  }

  test("word with duplicated character in mixed case") {
    
    val result = Isogram.isIsogram("Alphabet")
    result should be (false)
  }

  test("hypothetical isogrammic word with hyphen") {
    
    val result = Isogram.isIsogram("thumbscrew-japingly")
    result should be (true)
  }

  test("isogram with duplicated non letter character") {
    
    val result = Isogram.isIsogram("Hjelmqvist-Gryb-Zock-Pfund-Wax")
    result should be (true)
  }

  test("made-up name that is an isogram") {
    
    val result = Isogram.isIsogram("Emily Jung Schwartzkopf")
    result should be (true)
  }

}
